.github/workflows/release-go-task.yml

```diff
@@ -0,0 +1,71 @@
+# Source: https://github.com/arduino/tooling-project-assets/blob/main/workflow-templates/release-go-task.md
+name: Release
+
+env:
+  # As defined by the Taskfile's PROJECT_NAME variable
+  PROJECT_NAME: serial-discovery
+  # As defined by the Taskfile's DIST_DIR variable
+  DIST_DIR: dist
+  # The project's folder on Arduino's download server for uploading builds
+  AWS_PLUGIN_TARGET: /discovery/serial-discovery/
+  ARTIFACT_NAME: dist
+
+on:
+  push:
+    tags:
+      - "v[0-9]+.[0-9]+.[0-9]+*"
+
+jobs:
+  create-release-artifacts:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Create changelog
+        uses: arduino/create-changelog@v1
+        with:
+          tag-regex: '^[0-9]+\.[0-9]+\.[0-9]+.*$'
+          filter-regex: '^\[(skip|changelog)[ ,-](skip|changelog)\].*'
+          case-insensitive-regex: true
+          changelog-file-path: "${{ env.DIST_DIR }}/CHANGELOG.md"
+
+      - name: Install Task
+        uses: arduino/setup-task@v1
+        with:
+          repo-token: ${{ secrets.GITHUB_TOKEN }}
+          version: 3.x
+
+      - name: Build
+        run: task dist:all
+
+      - name: Identify Prerelease
+        # This is a workaround while waiting for create-release action
+        # to implement auto pre-release based on tag
+        id: prerelease
+        run: |
+          wget -q -P /tmp https://github.com/fsaintjacques/semver-tool/archive/3.0.0.zip
+          unzip -p /tmp/3.0.0.zip semver-tool-3.0.0/src/semver >/tmp/semver && chmod +x /tmp/semver
+          if [[ "$(/tmp/semver get prerel "${GITHUB_REF/refs\/tags\//}")" ]]; then echo "::set-output name=IS_PRE::true"; fi
+
+      - name: Create Github Release and upload artifacts
+        uses: ncipollo/release-action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          bodyFile: ${{ env.DIST_DIR }}/CHANGELOG.md
+          draft: false
+          prerelease: ${{ steps.prerelease.outputs.IS_PRE }}
+          artifacts: ${{ env.DIST_DIR }}/*
```

Comment:
```suggestion
          # NOTE: "Artifact is a directory" warnings are expected and don't indicate a problem
          # (all the files we need are in the DIST_DIR root)
          artifacts: ${{ env.DIST_DIR }}/*
```
The build output subfolders cause a bunch of warnings in [the workflow run summary](https://github.com/per1234/serial-discovery/actions/runs/1111406688) and [logs](https://github.com/per1234/serial-discovery/runs/3276512683?check_suite_focus=true#step:7:18):
```
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_Linux_32bit/. Directories can not be uploaded to a release.
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_Linux_64bit/. Directories can not be uploaded to a release.
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_Linux_ARM64/. Directories can not be uploaded to a release.
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_Linux_ARMv6/. Directories can not be uploaded to a release.
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_Linux_ARMv7/. Directories can not be uploaded to a release.
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_macOS_64bit/. Directories can not be uploaded to a release.
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_Windows_32bit/. Directories can not be uploaded to a release.
Warning: Artifact is a directory:dist/serial-discovery_0.0.42_Windows_64bit/. Directories can not be uploaded to a release.
```
Although there is no problem, I think these warnings could cause confusion far into the future when nobody remembers exactly how these workflows work.

I didn't find a clean solution for adjusting the `artifacts` input glob to exclude the subfolders so I settled on the comment.

I realize this is not in the "template" workflow, but I'm liking the approach of making adjustments downstream where the the change is less abstract and thus maybe easier for the reviewer to thoroughly evaluate, then pushing them back up to the source template if accepted.

I think the ideal solution would be to clean up these folders somewhere earlier in the build process (I guess  in DistTasks.yml right after generating the archive?), which would be especially beneficial for the full release workflow, since it passes them from job to job as workflow artifacts.

Reply:
> I think the ideal solution would be to clean up these folders somewhere earlier in the build process (I guess in DistTasks.yml right after generating the archive?), which would be especially beneficial for the full release workflow, since it passes them from job to job as workflow artifacts.

Yeah, I think this would probably be the best solution but am fine with the warning notice too for now.